client/src/layers/EditAnnotationLayer.ts

```diff
@@ -350,14 +399,21 @@ export default class EditAnnotationLayer extends BaseLayer<GeoJSON.Feature> {
     if (this.featureLayer === e.annotation.layer()) {
       if (e.annotation.state() === 'done' && this.formattedData.length === 0) {
         // geoJS insists on calling done multiple times, this will prevent that
```

Comment:
I think it might still be true, but belongs above the conditional.  I think that sometimes on initialization and in creation mode it would trigger a done multiple 'done' calls.  I just double checked and that comment is relevant but should be above the conditional.  It is why we check to make sure the `this.formattedData.length === 0`.  Which a more apt check might be to see use the `this.getMode() === 'creation'`.  What do you think about changing it to that?

Reply:
Yes, I agree with all of that.